mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -610,7 +627,9 @@ class GridOpInitPattern : public ConvertOpToLLVMPattern<Op>,
 
 /// Base pattern for operations returning the result of a SYCL grid operation
 /// \tparam GridOp.
-template <typename Op, typename GridOp>
+template <typename Op, typename GridOp, typename Getter,
+          typename = std::enable_if_t<
```

Comment:
This use of `std::enable_if` is discouraged by cppreference, see the section "Notes" [here](https://en.cppreference.com/w/cpp/types/enable_if).

Reply:
> A common mistake is to declare **two** function templates that differ only in their default template arguments.

This is only a problem if we declare two function templates. I think this usage is totally fine here. If you see example 5 in the "Example" section in the same page, you'll see this form is being used. I can swap to number 4 if you prefer, but I like this one more as I think it reads better